pkg/services/policer/policer.go

```diff
@@ -22,12 +23,41 @@ type nodeLoader interface {
 	ObjectServiceLoad() float64
 }
 
+type objectsInWork struct {
+	m    *sync.RWMutex
+	objs map[oid.Address]struct{}
+}
+
+func (oiw objectsInWork) inWork(addr oid.Address) bool {
+	oiw.m.RLock()
+	defer oiw.m.RUnlock()
+
+	_, ok := oiw.objs[addr]
+	return ok
+}
+
+func (oiw objectsInWork) remove(addr oid.Address) {
+	oiw.m.Lock()
+	defer oiw.m.Unlock()
```

Comment:
`defer` is redundant here, isn't it?

Reply:
do you mean to move it after accessing the map?

ok, usually prefer to work with mutexes via `defer` (do not know does it add any overhead in compiled code)